src/main.js

```diff
@@ -867,15 +871,30 @@ const CodingChatboxApplication = new Lang.Class({
     },
 
     vfunc_activate: function() {
-        if (!this._mainWindow)
-            this._mainWindow = new CodingChatboxMainWindow({
-                application: this,
-                actor_model: this._actorModel,
-                service: this._service,
-                game_service: this._gameService
+        if (!this._mainWindow) {
+            // Need to hold the application until we have
+            // a main window.
+            this.hold();
+
+            return Thumbnails.rewriteThumbnailersIntoDir(
+                CHATBOX_THUMBNAILERS_PATH,
+                null
+            ).then(() => {
```

Comment:
This hack was the best thing I could think of at the time that would convince libgnome-desktop allow the use of the host thumbnailers.

The better solution is probably to include the thumbnailers in the runtime. That said ... I wonde what would happen if we built them as part of the app's manifest.